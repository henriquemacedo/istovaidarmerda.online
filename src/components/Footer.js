import React from "react";
import Link from "next/link";
import styled from "styled-components";

const Wrapper = styled.footer`
  background-color: var(--lightBrown);
  padding: 100px 100px 50px 100px;
  color: var(--light);
`;

const FooterMenu = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(150px, 1fr));
  text-align: center;

  h4 {
    margin-bottom: 30px;
    text-transform: uppercase;
  }

  ul {
    margin: 0;
    padding: 0;

    li {
      list-style: none;

      &:not(:last-child) {
        margin-bottom: 10px;
      }

      a {
        color: var(--light);
        text-decoration: none;
        transition-duration: 0.3s;

        &:hover {
          color: var(--darkBrown);
        }
      }
    }
  }
`;

const Copyright = styled.div`
  margin-top: 50px;
  font-size: 0.85rem;
  text-align: center;
  color: var(--darkBrown);
`;

const Footer = () => (
  <Wrapper>
    <FooterMenu>
      <div>
        <h4>Contactos</h4>
        <ul>
          <li>
            <Link href="mailto:istovaidarmerda.online@gmail.com" passHref>
              <a target="_blank">Email</a>
            </Link>
          </li>
          <li>
            <Link href="https://twitter.com/SoniaTelesF" passHref>
              <a target="_blank">@SoniaTelesF</a>
            </Link>
          </li>
        </ul>
      </div>
      <div>
        <h4>Social</h4>
        <ul>
          <li>
            <Link href="https://twitter.com/IstoVai" passHref>
              <a target="_blank">Twitter</a>
            </Link>
          </li>
          <li>
            <Link href="#0" passHref>
              <a target="_blank">Youtube</a>
            </Link>
          </li>
        </ul>
      </div>
      <div>
        <h4>Media</h4>
        <ul>
          <li>
            <Link href="#0" passHref>
              <a target="_blank">Press Kit</a>
            </Link>
          </li>
        </ul>
      </div>
    </FooterMenu>
    <Copyright>{["Copyright Â© ", new Date().getFullYear(), " Isto Vai Dar Merda. Toda a merda reservada."]}</Copyright>
  </Wrapper>
);

export default Footer;
