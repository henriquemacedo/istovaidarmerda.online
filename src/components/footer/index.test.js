import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Footer from "./index";

describe("components/footer", () => {
  describe("render", () => {
    it("should render without errors", () => {
      const component = render(<Footer />);

      expect(component).toMatchSnapshot();
    });
  });
});
