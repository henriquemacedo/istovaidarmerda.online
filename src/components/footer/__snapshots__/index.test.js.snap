// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/footer render should render without errors 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <footer
        class="styles__Wrapper-sc-2nn9gf-0 cTxcED"
      >
        <div
          class="styles__FooterMenu-sc-2nn9gf-1 eWqYYe"
        >
          <div>
            <h4>
              Contactos
            </h4>
            <ul>
              <li>
                <a
                  href="mailto:istovaidarmerda.online@gmail.com"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Email
                </a>
              </li>
              <li>
                <a
                  href="https://twitter.com/SoniaTelesF"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  @SoniaTelesF
                </a>
              </li>
            </ul>
          </div>
          <div>
            <h4>
              Social
            </h4>
            <ul>
              <li>
                <a
                  href="https://twitter.com/IstoVai"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Twitter
                </a>
              </li>
              <li>
                <a
                  href="https://www.youtube.com/channel/UC-M0MHRa0CNewKhjCz4c2Qg"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Youtube
                </a>
              </li>
            </ul>
          </div>
          <div>
            <h4>
              Media
            </h4>
            <ul>
              <li>
                <a
                  href="https://drive.google.com/drive/folders/1J4i9QUIF_M7-OQ8qXOPmOJdGK3O7HUqR?usp=sharing"
                  target="_blank"
                >
                  Press Kit
                </a>
              </li>
            </ul>
          </div>
        </div>
        <div
          class="styles__Copyright-sc-2nn9gf-2 jTqdAj"
        >
          Copyright © 
          2020
           Isto Vai Dar Merda. Toda a merda reservada.
        </div>
      </footer>
    </div>
  </body>,
  "container": <div>
    <footer
      class="styles__Wrapper-sc-2nn9gf-0 cTxcED"
    >
      <div
        class="styles__FooterMenu-sc-2nn9gf-1 eWqYYe"
      >
        <div>
          <h4>
            Contactos
          </h4>
          <ul>
            <li>
              <a
                href="mailto:istovaidarmerda.online@gmail.com"
                rel="noopener noreferrer"
                target="_blank"
              >
                Email
              </a>
            </li>
            <li>
              <a
                href="https://twitter.com/SoniaTelesF"
                rel="noopener noreferrer"
                target="_blank"
              >
                @SoniaTelesF
              </a>
            </li>
          </ul>
        </div>
        <div>
          <h4>
            Social
          </h4>
          <ul>
            <li>
              <a
                href="https://twitter.com/IstoVai"
                rel="noopener noreferrer"
                target="_blank"
              >
                Twitter
              </a>
            </li>
            <li>
              <a
                href="https://www.youtube.com/channel/UC-M0MHRa0CNewKhjCz4c2Qg"
                rel="noopener noreferrer"
                target="_blank"
              >
                Youtube
              </a>
            </li>
          </ul>
        </div>
        <div>
          <h4>
            Media
          </h4>
          <ul>
            <li>
              <a
                href="https://drive.google.com/drive/folders/1J4i9QUIF_M7-OQ8qXOPmOJdGK3O7HUqR?usp=sharing"
                target="_blank"
              >
                Press Kit
              </a>
            </li>
          </ul>
        </div>
      </div>
      <div
        class="styles__Copyright-sc-2nn9gf-2 jTqdAj"
      >
        Copyright © 
        2020
         Isto Vai Dar Merda. Toda a merda reservada.
      </div>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
